src/components/global/Dialog/Dialog.tsx

```diff
@@ -0,0 +1,60 @@
+import { ReactEventHandler, RefObject, useEffect, useRef } from "react";
+import { useDialogPolyfill } from "./useDialogPolyfill";
```

Comment:
just a question : why we have useDialogPolyfill in separate file, but useDialogOpening and useDialogClosing here?
do we really need them to be a separate functions, as they use only useEffect and are used in one file?